### DISCUSSION TITLE
FINN with ONNX Exported from Tensorflow

### DISCUSSION THREAD

**Discussion by jomar0** [Xilinx/finn#781] [Q&A] [answered]
Hi,
My MSc Project involves creating a NN for Super Resolution and deploying it to an FPGA, however, I don't believe Brevitas supports quantisation-aware training for the layers I'll be using, namely Depthwise-seperable convolution, deconvolution and PixelShuffle, however, TensorFlow seemingly does. Is FINN limited to just Brevitas ONNX exports, or will it work with exports from TensorFlow. Will FINN also support these layers?
Thanks for the help!

**Comment by auphelia** [comment, marked as answer]
Hi @jomar0, 

FINN expects a certain flavor of ONNX (either [QONNX](https://github.com/fastmachinelearning/qonnx) or FINN-ONNX, see [here](https://finn.readthedocs.io/en/latest/internals.html#intermediate-representation-qonnx-and-finn-onnx)) to represent the quantization in the ONNX model. Brevitas provides exports to both of these formats.

Regarding your network: depthwise-separable convolutions and deconvolution are supported in Brevitas and PixelShuffle is a data reordering operator,  so it shouldn't require special support. 

But please be aware, that we currently don't have support for deconvolution in the FINN compiler, you could consider replacing your deconvolution nodes to nearest-neighbor upsampling together with a regular convolution, which is supported also from the compiler side.
